bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/discovery/MatterDiscoveryService.java

```diff
@@ -0,0 +1,160 @@
+    private @Nullable ScheduledFuture<?> discoveryJob;
+    private static final int REFRESH_INTERVAL = 60 * 5;
+
+    public MatterDiscoveryService() throws IllegalArgumentException {
+        // set a 2 min timeout, which should be plenty of time to discover devices, but stopScan will be called when the
+        // Matter client is done looking for new Nodes/Endpoints
+        super(Set.of(THING_TYPE_NODE, THING_TYPE_ENDPOINT), REFRESH_INTERVAL);
+    }
+
+    @Override
+    public void setThingHandler(ThingHandler handler) {
+        logger.debug("setThingHandler {}", handler);
+        if (handler instanceof MatterDiscoveryHandler childDiscoveryHandler) {
+            childDiscoveryHandler.setDiscoveryService(this);
+            this.thingHandler = handler;
+        }
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return thingHandler;
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        stopBackgroundDiscovery();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Start background discovery");
+        ScheduledFuture<?> discoveryJob = this.discoveryJob;
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            this.discoveryJob = scheduler.scheduleWithFixedDelay(this::startScan, REFRESH_INTERVAL, REFRESH_INTERVAL,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stop background discovery");
+        ScheduledFuture<?> discoveryJob = this.discoveryJob;
+        if (discoveryJob != null) {
+            discoveryJob.cancel(true);
+            this.discoveryJob = null;
+        }
+    }
+
+    @Override
+    public @Nullable String getScanInputLabel() {
+        return "@text/discovery.matter.scan-input.label";
+    }
+
+    @Override
+    public @Nullable String getScanInputDescription() {
+        return "@text/discovery.matter.scan-input.description";
+    }
+
+    @Override
+    protected void startScan() {
+        startScan("");
+    }
+
+    @Override
+    public void startScan(String input) {
+        ThingHandler handler = this.thingHandler;
+        if (handler != null && handler instanceof MatterDiscoveryHandler childDiscoveryHandler) {
+            childDiscoveryHandler.startScan(input.length() > 0 ? input : null).whenComplete((value, e) -> {
+                logger.debug("startScan complete");
+                stopScan();
+            });
+        }
+    }
+
+    public void discoverBridgeEndpoint(ThingUID thingUID, ThingUID bridgeUID, Endpoint root) {
+        String label = ("@text/discovery.matter.bridge-endpoint.label : "
+                + MatterLabelUtils.labelForBridgeEndpoint(root)).trim();
+        discoverThing(thingUID, bridgeUID, root, root.number.toString(), "endpointId", label);
```

Comment:
I have a big doubt that the label will be properly built.
Did you checked that it work well ?

Same remark for discoverNodeDevice.